bindings_generator/src/methods.rs

```diff
@@ -3,126 +3,118 @@ use crate::documentation::class_doc_link;
-        let MethodName {
-            rust_name: method_name,
-            ..
-        } = method.get_name();
-        if method_name == "free" {
-            continue;
-        }
-        writeln!(
-            output,
-            "            {}: 0 as *mut sys::godot_method_bind,",
-            method_name,
-        )?;
-    }
+    let has_underscore = api.api_underscore.contains(&class.name);
+
+    let method_table = format_ident!("{}MethodTable", class.name);
     let lookup_name: String = if has_underscore {
-        format!("_{class}", class = class.name)
+        format!("_{class}\0", class = class.name)
     } else {
-        class.name.clone()
+        format!("{}\0", class.name)
     };
-    writeln!(
-        output,
-        r#"
-        }};
-
-        &mut TABLE
-    }}
-
-    #[inline]
-    pub fn get(gd_api: &GodotApi) -> &'static Self {{
-        unsafe {{
-            let table = Self::get_mut();
-            static INIT: Once = Once::new();
-            INIT.call_once(|| {{
-                {name}MethodTable::init(table, gd_api);
-            }});
-
-            table
-        }}
-    }}
-
-    #[inline(never)]
-    fn init(table: &mut Self, gd_api: &GodotApi) {{
-        unsafe {{
-            let class_name = b"{lookup_name}\0".as_ptr() as *const c_char;
-            table.class_constructor = (gd_api.godot_get_class_constructor)(class_name);"#,
-        name = class.name,
-        lookup_name = lookup_name,
-    )?;
-    for method in &class.methods {
+
+    let struct_methods = class.methods.iter().filter_map(|m| {
+        let rust_name = format_ident!("{}", m.get_name().rust_name);
+        if rust_name != "free" {
+            Some(quote! { pub #rust_name: *mut sys::godot_method_bind })
+        } else {
+            None
+        }
+    });
+
+    let struct_definition = quote! {
+        #[doc(hidden)]
+        #[allow(non_camel_case_types)]
+        pub(crate) struct #method_table {
+            pub class_constructor: sys::godot_class_constructor,
+            #(#struct_methods),*
+        }
+    };
+
+    let impl_methods = class.methods.iter().filter_map(|m| {
+        let rust_name = format_ident!("{}", m.get_name().rust_name);
+        if rust_name != "free" {
+            Some(quote! { #rust_name: 0 as *mut sys::godot_method_bind })
+        } else {
+            None
+        }
+    });
+
+    let init_methods = class.methods.iter().filter_map(|m| {
         let MethodName {
-            rust_name: method_name,
+            rust_name,
             original_name,
-        } = method.get_name();
-        if method_name == "free" {
-            continue;
+        } = m.get_name();
+
+        let rust_name = format_ident!("{}", rust_name);
+        let original_name = format!("{}\0", original_name);
+
+        if rust_name != "free" {
+            Some(quote! {
+                table.#rust_name = (gd_api.godot_method_bind_get_method)(class_name, #original_name.as_ptr() as *const c_char );
```

Comment:
I know it was just defined up a few lines but maybe if we depend on strings containing a NUL at the end then maybe we can put an `assert!` before the `quote!` invocations?

Obviously you noticed this during the migration but it could have been easy for me (as a reviewer I mean) to miss, so maybe asserts might be nice as a kind of future-proofing?